src/Modules/OrchardCore.Commerce/Services/StripePaymentService.cs

```diff
@@ -278,6 +286,49 @@ public async Task UpdateOrderToPaymentFailedAsync(PaymentIntent paymentIntent)
         return order;
     }
 
+    public async Task<ContentItem> CreateNoPaymentOrderFromShoppingCartAsync(IUpdateModelAccessor updateModelAccessor)
+    {
+        var currentShoppingCart = await _shoppingCartPersistence.RetrieveAsync();
+
+        var order = await _contentManager.NewAsync(Constants.ContentTypes.Order);
+        if (await UpdateOrderWithDriversAsync(order, updateModelAccessor))
+        {
+            return null;
+        }
+
+        var guid = Guid.NewGuid().ToString();
+        order.DisplayText = T["Order {0}", guid];
+
+        var lineItems = await CreateOrderLineItemsAsync(currentShoppingCart);
+
+        var cartViewModel = await _shoppingCartHelpers.CreateShoppingCartViewModelAsync(
+            shoppingCartId: null,
+            order.As<OrderPart>().ShippingAddress.Address,
+            order.As<OrderPart>().BillingAddress.Address);
+
+        order.Alter<OrderPart>(orderPart =>
+        {
+            // Shopping cart
+            orderPart.LineItems.Clear();
+            orderPart.LineItems.AddRange(lineItems);
+
+            orderPart.OrderId.Text = guid;
+            orderPart.Status = new TextField { ContentItem = order, Text = OrderStatuses.Pending.HtmlClassify() };
```

Comment:
You don't have to overwrite the whole field. `OrderPart.Status` is never null because it has a `new()` default value, so it's enough to just overwrite `orderPart.Status.Text`.